src/sqlite3/statement.cr

```diff
@@ -19,7 +19,11 @@ class SQLite3::Statement < DB::Statement
     end
 
     # exec
-    step = LibSQLite3::Code.new LibSQLite3.step(self)
+    step = nil
```

Comment:
`step` can be initialized with

```crystal
step = uninitialized LibSQLite3::Code
```

to avoid a union here. The loop will be executed at least one time for sure.

Reply:
Will do, thanks for the feedback!